### DISCUSSION TITLE
Annotations inside image captions

### DISCUSSION THREAD

**Discussion by MrEddX** [squidfunk/mkdocs-material#7357] [Q&A] [answered]
Is it possible to add annotations inside image captions?
If this is possible, could you provide an example?

**Comment by alexvoss** [comment]
I assume you use `<figure>` as [suggested in the documentation](https://squidfunk.github.io/mkdocs-material/reference/images/?h=caption#image-captions)? To get an annotation to render there you would need to add the `markdown` attribute to `<figcaption>` so that the content of that HTML element gets treated as Markdown.

**Comment by MrEddX** [comment]
Yes. You're right @alexvoss. But in my case the annotation icon is half visible so I'm not sure if I'm doing it the right way. For example:

```html
<figure markdown>
  ![Image](resources/img/image_light.png#only-light){ width="640" }
  ![Image](resources/img/image_dark.png#only-dark){ width="640" }
  <figcaption markdown>Image caption (1)  text here
  { .annotate }

  1. And the annotation text.

  </figcaption>
</figure>
```

| Light theme | Dark theme |
|--------------|---------------|
| ![image](https://github.com/user-attachments/assets/fc878f8a-c9ce-4302-9800-156c05ee5a12)  | ![image](https://github.com/user-attachments/assets/0d4863ff-0046-402f-9191-d6ef5bdaa655) |
| ![image](https://github.com/user-attachments/assets/06da7228-773f-4a27-a58a-c9582a7209f6) | ![image](https://github.com/user-attachments/assets/97f009c3-1afb-4db1-abc3-a354b5fa2bb6) |

When you click on the _half icon_, the animation works and the annotation is displayed, just the rendering is weird.

**Reply by kamilkrzyskow** [reply, marked as answer]
Hi @MrEddX,
I think this is a fixable bug, please report it via the proper Issue channels with a minimal reproduction.
- https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/#creating-a-reproduction

For the time being you can add CSS customization:
- https://squidfunk.github.io/mkdocs-material/customization/#additional-css
```css
figure .md-annotation {
  text-align: initial;
}
```

**Reply by kamilkrzyskow** [reply]
I see this was resolved in:
- https://github.com/squidfunk/mkdocs-material/issues/7362

Thanks for reporting. Also due to how the `{ .annotate }` syntax works, I believe this a valid alternative approach, which is a bit cleaner:
```md
<figcaption class="annotate" markdown>

Image caption (1) text here

</figcaption>

1. And the annotation text.
```

However, as it moves the list outside of the figcaption, then perhaps the position of the list shifts during printing.

**Comment by MrEddX** [comment]
Thanks to everyone who helped resolve this situation! Tested with the new [9.5.29](https://github.com/squidfunk/mkdocs-material/releases/tag/9.5.29) and now it works as expected. 

I like your way @kamilkrzyskow. It's much more elegant.
